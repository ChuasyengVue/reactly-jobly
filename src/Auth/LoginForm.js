import React, { useState } from "react";
import { useNavigate } from "react-router-dom";

const LoginForm = ({ login }) => {
    const navigate = useNavigate();

    const[formData, setFormData] = useState({
        username:"",
        password:""
    });

    const handleChange = (evt) =>{
        const {name, value} = evt.target;
        setFormData(formData => ({...formData, [name]:value }));
    }

    const handleSubmit = async (evt) =>{
        evt.prevenDefault();
        try {
            await login(formData)
            navigate('/');
        } catch (error) {
            console.error("Login failed:", error);
        }
    }

    return(
        <div>
            <form onSubmit={handleSubmit}>
                <h3>Log In</h3>
                <div>
                    <label>Username</label>
                    <input 
                    name="username"
                    value={formData.username}
                    onChange={handleChange}
                    required
                    />
                </div>

                <div>
                    <label>Password</label>
                    <input
                    name="password"
                    type="password"
                    value={formData.password}
                    onChange={handleChange}
                    required
                    />
                </div>

                <button type='submit'> Submit </button>
            </form>
        </div>
    )

}

export default LoginForm;