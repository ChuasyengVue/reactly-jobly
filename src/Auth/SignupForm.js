import React, { useState } from "react";
import { useNavigate } from 'react-router-dom';

const Signup = ({ signup }) => {

    const navigate = useNavigate();

    const[formData, setFormData] = useState({
        username:"",
        password:"",
        firstName:"",
        lastName:"",
        email:""
    });

    

    const handleChange = (evt) => {
        const {name, value} = evt.target;
        setFormData(prevFormData => ({
            ...prevFormData, [name]:value
        }));
    }

    const handleSubmit = async(evt) => {
        evt.preventDefault();

        try {
            await signup(formData);
            navigate('/');  //Redirects to homepage if sign up success.
        } catch (error) {
            console.error("Sign up failed:" , error);
        }

    }

    return(
        <div>
            
            <form onSubmit={handleSubmit}>
                <h3>Sign Up</h3>
                <div>
                    <label>Username</label>
                    <input
                    name="username"
                    type="text"
                    value={formData.username}
                    onChange={handleChange}
                    required
                    />
                </div>

                <div>
                    <label>Password</label>
                    <input
                    name="password"
                    type="password"
                    value={formData.password}
                    onChange={handleChange}
                    required
                    />
                </div>

                <div>
                    <label>First Name</label>
                    <input
                    name="firstName"
                    type="text"
                    value={formData.firstName}
                    onChange={handleChange}
                    required
                    />
                </div>

                <div>
                    <label>Last Name</label>
                    <input
                    name="lastName"
                    type="text"
                    value={formData.lastName}
                    onChange={handleChange}
                    required
                    />
                </div>

                <div>
                    <label>Email</label>
                    <input
                    name="email"
                    type="email"
                    value={formData.email}
                    onChange={handleChange}
                    required
                    />
                </div>

                <button type="submit">Submit</button>

            </form>
        </div>
    )
}

export default Signup;