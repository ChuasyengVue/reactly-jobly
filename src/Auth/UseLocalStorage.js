// import { useState } from "react";

// function useLocalStorage(key, initialValue) {
//     const [storedValue, setStoredValue] = useState(() => {
//         try {
//             const item = window.localStorage.getItem(key);
//             return item ? JSON.parse(item) : initialValue;
//         } catch (error) {
//             console.error(error);
//             return initialValue;
//         }
//     });

//     const setValue = (value) => {
//         try {
//             const valueToStore = value instanceof Function ? value(storedValue) : value;
//             setStoredValue(valueToStore);
//             window.localStorage.setItem(key, JSON.stringify(valueToStore));
//         } catch (error) {
//             console.error(error);
//         }
//     };

//     return [storedValue, setValue];
// }

// export default useLocalStorage;  // Make sure this is a default export


import { useState, useEffect } from "react";

/** Custom hook for keeping state data synced with localStorage.
 *
 * This creates `item` as state and look in localStorage for current value
 * (if not found, defaults to `firstValue`)
 *
 * When `item` changes, effect re-runs:
 * - if new state is null, removes from localStorage
 * - else, updates localStorage
 *
 * To the component, this just acts like state that is also synced to/from
 * localStorage::
 *
 *   const [myThing, setMyThing] = useLocalStorage("myThing")
 */

function useLocalStorage(key, firstValue = null) {
  const initialValue = localStorage.getItem(key) || firstValue;

  const [item, setItem] = useState(initialValue);

  useEffect(function setKeyInLocalStorage() {
    console.debug("hooks useLocalStorage useEffect", "item=", item);

    if (item === null) {
      localStorage.removeItem(key);
    } else {
      localStorage.setItem(key, item);
    }
  }, [key, item]);

  return [item, setItem];
}

export default useLocalStorage;
