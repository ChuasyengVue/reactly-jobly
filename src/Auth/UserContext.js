import  { createContext, useState } from 'react';

export const UserContext = createContext({
    currentUser: null,
    login: () => {},
    signup: () => {},
    logout: () => {},
});

export const UserProvider = ({ children }) => {
    const [currentUser, setCurrentUser] = useState(null);

    const updateUser = (updatedUser) => {
        setCurrentUser((prevUser) => ({
            ...prevUser,
            ...updateUser,
        }));
    }

    return(
        <UserContext.Provider value={{currentUser, updateUser }}>
            {children}
        </UserContext.Provider>
    )
}