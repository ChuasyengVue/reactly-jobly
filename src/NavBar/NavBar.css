.navbar {
    background-color: white; /* Dark background */
    padding: 1px 5px; /* Padding for the navbar */

    .navbar-home {
        color: #f4bb3a; /* Brand color */
        font-size: 1.5rem; /* Larger font for brand */
        text-decoration: none;
        
        display: flex;
        &:hover {
            color: #ed7b32; /* Change color on hover */
        }
    }

    .nav-item {
        color: gray; /* Text color */
        margin: 0 20px; /* Spacing between items */
        text-decoration: none;
        padding: 1px;
        
        &:hover {
            color: #f4bb3a; /* Change color on hover */
        }
    }
}
