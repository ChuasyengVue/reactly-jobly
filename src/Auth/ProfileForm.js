import React, { useState, useContext, useEffect } from "react";
import JoblyApi from "../Api/Api";
import { UserContext } from "./UserContext";


const ProfileForm = () => {
  
    const {currentUser, updateUser} = useContext(UserContext);

  const [formData, setFormData] = useState({
    username: currentUser.username,
    firstName: currentUser.firstName,
    lastName: currentUser.lastName,
    email: currentUser.email,
    password: "",
  });

  useEffect(() => {
    if(currentUser) {
        setFormData({
            
            firstName:currentUser.firstName || '',
            lastName:currentUser.lastName || '',
            email:currentUser.email || '',
            password:currentUser.password || ''
        });
    }
  },[currentUser]);

  async function handleSubmit(evt) {
    evt.preventDefault();
    try {
      const updatedUser = await JoblyApi.updateProfile(currentUser.username, formData); // Update user profile
      updateUser(updatedUser);
      alert("Profile updated successfully!");
    } catch (err) {
      console.error("Profile update failed", err);
      
    }
  }

  async function handleChange(evt) {

    const {name, value} = evt.target;
    setFormData(formData => ({...formData, [name]:value}));
  }

  return (
    <form onSubmit={handleSubmit}>
    <div>
      <label>Username</label>
      <input name="username" value={formData.username} disabled />
      </div>
      <div>
      <label>First Name</label>
      <input
        name="firstName"
        value={formData.firstName}
        onChange={handleChange}
      />
    </div>
    <div>
      <label>Last Name</label>
      <input
        name="lastName"
        value={formData.lastName}
        onChange={handleChange}
      />
    </div>
    <div>
      <label>Email</label>
      <input
        name="email"
        value={formData.email}
        onChange={handleChange}
      />
    </div>
    <div>
      <label>Password</label>
      <input
        name="password"
        type="password"
        value={formData.password}
        onChange={handleChange}
      />
      </div>

      <button type="submit">Save Changes</button>
    </form>
  );
};

export default ProfileForm;
