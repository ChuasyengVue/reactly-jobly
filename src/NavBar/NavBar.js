import React, {useContext} from "react";
import { NavLink } from 'react-router-dom';
import { Navbar, Nav } from 'reactstrap';
import { UserContext } from "../Auth/UserContext";
import './NavBar.css';

const NavBar = ({logout}) => {

    const {currentUser} = useContext(UserContext);

    // If user is logged in, show companies/jobs/profile/logout links
    function LoggedInNav() {
        return(
            <Nav className="ml-auto">
                <NavLink className="nav-item" to="/companies" >Companies</NavLink>
                <NavLink className="nav-item" to="/jobs" >Jobs</NavLink>
                <NavLink className="nav-item" to="/profile" >Profile</NavLink>
                <NavLink className="nav-item" to='/' onClick={logout}>Logout {currentUser.username}</NavLink>
            </Nav>
        )
    }

    // If user is not logged in, show login/signup
    function LoggedOutNav() {
        return(
            <Nav>
                <NavLink className="nav-item" to="/login" >Login</NavLink>
                <NavLink className="nav-item" to="/signup" >Sign Up</NavLink>
            </Nav>
        )
    }
    
    return(
        <Navbar className="navbar" expand="md">
            <NavLink className="navbar-home"to='/'>Jobly</NavLink>
           {currentUser ? LoggedInNav() : LoggedOutNav()}
        </Navbar>
    )
}
export default NavBar;