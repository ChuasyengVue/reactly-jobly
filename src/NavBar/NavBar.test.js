import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import NavBar from "./NavBar";
import { UserProvider } from "../Auth/UserContext";

//smoke test
it("renders without crashing", function() {
    render(
        <MemoryRouter>
            <NavBar />
        </MemoryRouter>
    );
});

// snapshot
it('it matches snapshot', function() {
    const { asFragment } = render(
        <MemoryRouter>
            <NavBar />
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

it("it matches snapshot when logged out", function () {
    const { asFragment } = render(
        <MemoryRouter>
          <UserProvider currentUser={null}>
            <NavBar />
          </UserProvider>
        </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  