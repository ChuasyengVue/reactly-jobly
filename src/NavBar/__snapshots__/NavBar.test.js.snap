// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar navbar-expand-md"
  >
    <div
      class="container-fluid"
    >
      <a
        aria-current="page"
        class="navbar-home active"
        href="/"
      >
        Jobly
      </a>
      <ul
        class="nav"
      >
        <a
          class="nav-item"
          href="/login"
        >
          Login
        </a>
        <a
          class="nav-item"
          href="/signup"
        >
          Sign Up
        </a>
      </ul>
    </div>
  </nav>
</DocumentFragment>
`;

exports[`it matches snapshot when logged out 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar navbar-expand-md"
  >
    <div
      class="container-fluid"
    >
      <a
        aria-current="page"
        class="navbar-home active"
        href="/"
      >
        Jobly
      </a>
      <ul
        class="nav"
      >
        <a
          class="nav-item"
          href="/login"
        >
          Login
        </a>
        <a
          class="nav-item"
          href="/signup"
        >
          Sign Up
        </a>
      </ul>
    </div>
  </nav>
</DocumentFragment>
`;
